tasks/go.py

```diff
@@ -403,13 +403,46 @@ def generate_licenses(ctx, filename='LICENSE-3rdparty.csv', verbose=False):
 def generate_protobuf(ctx):
     """
     Generates protobuf defintions in pkg/proto
+
+    We must build the packages one at a time due to protoc-gen-go limitations
     """
+
+    # Key: path, Value: grpc_gateway
+    PROTO_PACKAGES = {
+        'model/v1': False,
+        'remoteconfig': False,
+        'api/v1': True,
+        'trace': False,
+        'trace/otlp': True,
+        'process': False,
+    }
+
+    # maybe put this in a separate function
+    PACKAGE_PLUGINS = {
+        'trace': '--gogoslick_out=',
+    }
+
+    # msgp targets
+    msgp_targets = {
+        'trace': ['remote_rates.pb.go', 'span.pb.go', 'stats.pb.go', 'tracer_payload.pb.go', 'trace.go'],
+        'core': ['remoteconfig.pb.go'],
+    }
+
     base = os.path.dirname(os.path.abspath(__file__))
     repo_root = os.path.abspath(os.path.join(base, ".."))
     proto_root = os.path.join(repo_root, "pkg", "proto")
+    protodep_root = os.path.join(proto_root, "protodep")
 
     print(f"nuking old definitions at: {proto_root}")
-    file_list = glob.glob(os.path.join(proto_root, "pbgo", "*.go"))
+    file_list = glob.glob(os.path.join(proto_root, "pbgo", "*.pb.go"))
+    for file_path in file_list:
+        try:
+            os.remove(file_path)
+        except OSError:
+            print("Error while deleting file : ", file_path)
```

Comment:
Punctuation usually goes without space before, and the space after is added automatically by Python.

```suggestion
            print("Error while deleting file:", file_path)
```